modules/accumulo/src/main/java/org/apache/fluo/accumulo/data/MutableBytes.java

```diff
@@ -91,6 +92,11 @@ public MutableBytes(String s, Charset cs) {
   }
 
   @Override
+  public final int hashCode() {
+    return Arrays.hashCode(data);
```

Comment:
I realized this will not work, because it hashes the entire array.  Its possible for a Bytes object to use a subsequence of an array.   In this case the hash code must be computed using the subsequence.  If its not then two Bytes objects that have equal content can have different hash codes, which is very bad.  The example below shows this.

```java
    Bytes b1 = Bytes.of("abcde");
    Bytes b2 = Bytes.of("cde");

    // ensure the hascode is correctly computed for a subsequence
   if(b2.hashCode() != b1.subSequence(2, 5).hashCode())){
      System.err.println("oh no");
   }
```

I looked and Arrays does not have a hashCode method that takes an offset and length, so I don't see a way to use Arrays w/o copying.  I think it would be best to revert the changes to MutableBytes and reintroduce the old hashCode function in Bytes.  Also the old hashCode function had caching behavior that would be nice to keep.  Can you also add a test to BytesTest to check that the hashCode of a subsequence works properly?